client/core/core.go

```diff
@@ -7001,14 +7095,36 @@ func (c *Core) authDEX(dc *dexConnection) error {
 		localBondMap[bondKey(dbBond.AssetID, dbBond.CoinID)] = struct{}{}
 	}
 
+	var unknownBondStrength uint32
+	unknownBondAssetID := -1
 	for _, bond := range result.ActiveBonds {
 		key := bondKey(bond.AssetID, bond.CoinID)
 		if _, found := localBondMap[key]; found {
 			continue
 		}
-		symb := dex.BipIDSymbol(bond.AssetID)
-		bondIDStr := coinIDString(bond.AssetID, bond.CoinID)
-		c.log.Warnf("Unknown bond reported by server: %v (%s)", bondIDStr, symb)
+		// Server reported a bond we do not know about.
+		ubs, ubaID := c.findBond(dc, bond)
+		unknownBondStrength += ubs
+		if unknownBondAssetID != -1 && ubs != 0 && uint32(unknownBondAssetID) != ubaID {
+			c.log.Warnf("Found unknown bonds for different assets. %s and %s.",
+				unbip(uint32(unknownBondAssetID)), unbip(ubaID))
+		}
+		if ubs != 0 {
+			unknownBondAssetID = int(ubaID)
+		}
+	}
+
+	// If there were unknown bonds, and target tier is zero, assume this was
+	// a restored account and set the tier according to the previous tier.
+	if unknownBondStrength > 0 {
+		dc.acct.authMtx.RLock()
+		if dc.acct.targetTier == 0 {
+			c.log.Infof("Setting target tier with host %s to %d with asset %s.",
+				dc.acct.host, unknownBondStrength, unbip(uint32(unknownBondAssetID)))
+			dc.acct.targetTier = uint64(unknownBondStrength)
+			dc.acct.bondAsset = uint32(unknownBondAssetID)
+		}
+		dc.acct.authMtx.RUnlock()
```

Comment:
On the fence on this move. It's really risky to assume target tier.

Reply:
Do we have a system for a random alert dialog where the user can specify? Or should I not set and send a notification?

Reply:
Notification sounds pretty good to start. Can work on better gui later.